eZ/Publish/Core/FieldType/RichText/XSLTProcessorExtension.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace eZ\Publish\Core\FieldType\RichText;
+
+/**
+ * Helper functions available in XSL documents.
+ */
+class XSLTProcessorExtension
+{
+    const URL_PATTERN = '%^((ezlocation|ezurl|ezcontent):\/\/\d+(#[^\s]+)?)|(#[^\s]+)|(?:(?:https?|ftp)://)(?:\S+(?::\S*)?@|\d{1,3}(?:\.\d{1,3}){3}|(?:(?:[a-z\d\x{00a1}-\x{ffff}]+-?)*[a-z\d\x{00a1}-\x{ffff}]+)(?:\.(?:[a-z\d\x{00a1}-\x{ffff}]+-?)*[a-z\d\x{00a1}-\x{ffff}]+)*(?:\.[a-z\x{00a1}-\x{ffff}]{2,6}))(?::\d+)?(?:[^\s]*)?$%iu';
```

Comment:
If this is based on what I used in https://github.com/ezsystems/ezpublish-kernel-ee/pull/161/files#diff-981d67ab6f6be0f1d1ff989637c9e357R82 I think it would be good to reference where it comes from, like I did there, just to give them credit.

Reply:
I've added note about regex author in docblock on `isValidUrl` method